packages/cms-lib/lib/config.js

```diff
@@ -77,14 +79,77 @@ const getOrderedConfig = unorderedConfig => {
   };
 };
 
-const writeConfig = () => {
-  logger.debug(`Writing current config to ${_configPath}`);
-  fs.writeFileSync(
-    _configPath,
-    yaml.safeDump(
-      JSON.parse(JSON.stringify(getOrderedConfig(_config), null, 2))
-    )
-  );
+const charToGlob = ch => `[${ch.toUpperCase()}${ch.toLowerCase()}]`;
+const wordToGlobs = word => [].map.call(word, charToGlob).join('');
+
+// .gitignore can be case-sensitive (configurable by user)
+// const CONFIG_GITIGNORE_PATTERN = 'hubspot.config.*';
+const CONFIG_GITIGNORE_PATTERN = `${wordToGlobs('hubspot')}.${wordToGlobs(
+  'config'
+)}.*`;
+const CONFIG_GITIGNORE_PATTERN_REGEX = /hubspot\.config\..*/i;
+
+/**
+ * @param {object}  options
+ * @param {string}  options.path
+ */
+const ensureConfigGitignore = async (options = {}) => {
+  const configPath = options.path || _configPath;
+  try {
+    const dir = path.dirname(configPath);
+    const ignoreFilePath = path.join(dir, '.gitignore');
+    await fs.ensureFile(ignoreFilePath);
+    let source = (await fs.readFile(ignoreFilePath)) || '';
+    if (source) {
+      const lines = source.split(/[\n\r]/);
+      const hasPattern =
+        lines.find(
+          CONFIG_GITIGNORE_PATTERN_REGEX.test,
+          CONFIG_GITIGNORE_PATTERN_REGEX
+        ) != null;
```

Comment:
I think it may be possible to use https://www.npmjs.com/package/ignore which we are already using for `.hsignore` for this.